L1Trigger/L1TMuonEndCap/src/TTPrimitiveConversion.cc

```diff
@@ -0,0 +1,133 @@
+
+#include "L1Trigger/L1TMuonEndCap/interface/SectorProcessorLUT.h"
+#include "L1Trigger/L1TMuonEndCap/interface/TrackTools.h"
+
+
+void TTPrimitiveConversion::configure(
+    const TTGeometryTranslator* tp_ttgeom,
+    const SectorProcessorLUT* lut,
+    int verbose, int endcap, int sector, int bx
+) {
+  assert(tp_ttgeom != nullptr);
+  assert(lut != nullptr);
+
+  tp_ttgeom_ = tp_ttgeom;
+  lut_       = lut;  // not used
+
+  verbose_ = verbose;
+  endcap_  = endcap; // 1 for ME+, 2 for ME-
+  sector_  = sector;
+  bx_      = bx;
+}
+
+void TTPrimitiveConversion::process(
+    const std::map<int, TTTriggerPrimitiveCollection>& selected_ttprim_map,
+    EMTFHitCollection& conv_hits
+) const {
+  std::map<int, TTTriggerPrimitiveCollection>::const_iterator map_tp_it  = selected_ttprim_map.begin();
+  std::map<int, TTTriggerPrimitiveCollection>::const_iterator map_tp_end = selected_ttprim_map.end();
+
+  for (; map_tp_it != map_tp_end; ++map_tp_it) {
+    TTTriggerPrimitiveCollection::const_iterator tp_it  = map_tp_it->second.begin();
+    TTTriggerPrimitiveCollection::const_iterator tp_end = map_tp_it->second.end();
+
+    for (; tp_it != tp_end; ++tp_it) {
+      EMTFHit conv_hit;
+      convert_tt(*tp_it, conv_hit);
+      conv_hits.push_back(conv_hit);
+    }
+  }
+}
+
+void TTPrimitiveConversion::process_no_prim_sel(
+    const TTTriggerPrimitiveCollection& ttmuon_primitives,
+    EMTFHitCollection& conv_hits
+) const {
+  TTTriggerPrimitiveCollection::const_iterator tp_it  = ttmuon_primitives.begin();
+  TTTriggerPrimitiveCollection::const_iterator tp_end = ttmuon_primitives.end();
+
+  for (; tp_it != tp_end; ++tp_it) {
+    if (endcap_ == 1 && sector_ == 1 && bx_ == tp_it->getTTData().bx) {  //FIXME: stupidly put everything into sector +1, to be fixed.
+      EMTFHit conv_hit;
+      convert_tt(*tp_it, conv_hit);
+      conv_hits.push_back(conv_hit);
+    }
+  }
+}
+
+
+// _____________________________________________________________________________
+// TT functions
+void TTPrimitiveConversion::convert_tt(
+    const TTTriggerPrimitive& ttmuon_primitive,
+    EMTFHit& conv_hit
+) const {
+  //const DetId&   tp_detId = ttmuon_primitive.detId();
+  const TTData&  tp_data  = ttmuon_primitive.getTTData();
+
+  int tp_region    = tp_ttgeom_->region(ttmuon_primitive);     // 0 for Barrel, +/-1 for +/- Endcap
+  int tp_endcap    = (tp_region == -1) ? 2 : tp_region;
+  int tp_station   = tp_ttgeom_->layer(ttmuon_primitive);
+  int tp_ring      = tp_ttgeom_->ring(ttmuon_primitive);
+  int tp_chamber   = tp_ttgeom_->module(ttmuon_primitive);
+  int tp_sector    = 1;  //FIXME
+  int tp_subsector = 0;  //FIXME
+
+  const bool is_neighbor = false;  //FIXME
+
+  // Set properties
+  //conv_hit.SetTTDetId      ( tp_detId );
+
+  conv_hit.set_endcap      ( (tp_endcap == 2) ? -1 : tp_endcap );
+  conv_hit.set_station     ( tp_station );
+  conv_hit.set_ring        ( tp_ring );
+  //conv_hit.set_roll        ( tp_roll );
```

Comment:
Remove commented-out lines.

Reply:
I keep these lines to list all the possible setters provided by the EMTFHit class. This makes it easier to remember to set it every time a new setter is added to the class. As you can just CTRL+F one of the existing setters, and just duplicate the line. I think it's a good practice.

Reply:
Fair enough